tests/Test/Proto/Files.hs

```diff
@@ -0,0 +1,560 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE DeriveAnyClass #-}
+{-# LANGUAGE DeriveGeneric #-}
+{-# LANGUAGE GADTs #-}
+{-# LANGUAGE NamedFieldPuns #-}
+{-# LANGUAGE NegativeLiterals #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE TypeApplications #-}
+{-# LANGUAGE TypeOperators #-}
+{-# OPTIONS_GHC -fno-warn-missing-export-lists #-}
+{-# OPTIONS_GHC -fno-warn-name-shadowing #-}
+{-# OPTIONS_GHC -fno-warn-unused-imports #-}
+{-# OPTIONS_GHC -fno-warn-unused-matches #-}
+
+module Test.Proto.Files 
+  ( dotProtoEmptyField
+  , dotProtoImport
+  , dotProtoLeadingDot
+  , dotProtoNegativeEnum
+  , dotProtoNestedMessage
+  , dotProtoOneofImport
+  , dotProtoOneof
+  , dotProtoOptional
+  , dotProtoProtocPlugin
+  , dotProtoWrappers
+  , dotProtoCommon
+  ) where
+
+import Proto3.Suite.DotProto.AST (DotProto)
+
+import NeatInterpolation qualified as Neat
+
+import Test.Proto.Quote (dotProtoTestFile)
+
+--------------------------------------------------------------------------------
+
+dotProtoEmptyField :: DotProto
+dotProtoEmptyField = 
```

Comment:
How is knowing the `DotProto` sufficient to do encoding and decoding tests?

Reply:
I don't understand the question, sorry.